src/kmeds_pywrapper.cpp

```diff
@@ -38,8 +38,25 @@ class KMedsWrapper : public KMedoids {
    * @param k The number of medoids to compute
    * @param logFilename The name of the outputted log file
    */
-  void fitPython(py::array_t<double> inputData, std::string loss, int k, std::string logFilename) {
-    KMedoids::setNMedoids(k);
+  void fitPython(py::array_t<double> inputData, std::string loss, std::string logFilename, py::kwargs kw) {
+    // throw an error if the number of medoids is not specified in either the KMedoids object or the fitPython function
+    try {
+      if (!KMedoids::getNMedoids()) {
+        if (kw.size() == 0) {
+          throw py::value_error("n_medoids");
+        }
+      }
+    } catch (py::error_already_set &c) {
+       // py::value_error is not a Python exception
+      assert(false);
```

Comment:
What does this do?

Reply:
The error catch follows from an example in pybind11 exceptions document: https://pybind11.readthedocs.io/en/stable/advanced/exceptions.html#handling-exceptions-from-python-in-c

When C++ calls Python functions, such as in a callback function or when manipulating Python objects, and Python raises an Exception, pybind11 converts the Python exception into a C++ exception of type `pybind11::error_already_set ` whose payload contains a C++ string textual summary and the actual Python exception. `error_already_set` is used to propagate Python exception back to Python (or possibly, handle them in C++).

I have removed this part and the error message is still raised properly. It seems we don't need this part.